src/formatDate.js

```diff
@@ -50,7 +50,40 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formattedArray = [];
+
+  const dateInput = {
+
+    dataConverter(format, value) {
+      this[format] = value;
+
+      if (format === 'YYYY') {
+        this.YY = value.substring(value.length - 2);
+      } else {
+        this.YYYY = value < 30 ? `20${value}` : `19${value}`;
+      }
+    },
+
+    separator: fromFormat[fromFormat.length - 1],
+  };
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      dateInput.dataConverter(
+        fromFormat[i], date.split(dateInput.separator)[i]
+      );
+    } else {
+      dateInput[fromFormat[i]] = date.split(dateInput.separator)[i];
```

Comment:
Similar to line 25, store the split date in a variable and use it in the loop instead of splitting it again and again.